code/game/objects/items/weapons/grenades/custom_grenades.dm

```diff
@@ -303,3 +303,97 @@
 	new /obj/item/multitool(src) // used to adjust the chemgrenade's signaller
 	new /obj/item/assembly/signaler(src)
 	return
+
+// --------------------------------------
+//  BSA Signaller
+// --------------------------------------
+
+/obj/item/grenade/bsa
+	name = "bluespace artillery signal beacon"
+	desc = "Call in a one time strike with bluespace artillery. It is filled with various biometric security measures."
+	icon_state = "bsa_beacon"
+	item_state = "bsa_beacon"
+	req_one_access = list(access_cent_commander)
+	det_time= 8 //in seconds, we multiply by 10 below so we can use this var in the messages we send. so a value of 8 here would be a value of 80 for the timer.
+	var/arm_time = 40 //in deciseconds, so 40 would equal 4 seconds.
+	var/caller
+	var/centcom_cancast = FALSE //incase admin wants to varedit to allow on Z 2 for whatever reason.
+	var/ex_power = 3
+	var/authorized_special_roles = list(SPECIAL_ROLE_ERT, SPECIAL_ROLE_DEATHSQUAD)
+	var/ID_access_override = TRUE
+	var/require_mindshield = TRUE
+
+/obj/item/grenade/bsa/proc/authenticate(var/mob/user, var/mindshield_check, var/role_check, var/access_check)
+	//mindshield check
+	if(mindshield_check && !ismindshielded(user))
+		return FALSE
+	//access check
+	if(access_check && !allowed(user))
+		return FALSE
+	//special role check
+	if(role_check)
+		var/role = FALSE
+		for(var/R in authorized_special_roles)
+			if(user.mind.special_role == R)
+				role = TRUE
+				break
+		if(!role)
+			return FALSE
+	return TRUE
+
+/obj/item/grenade/bsa/attack_self()
+	var/mob/user = usr
+	var/turf/bombturf = get_turf(src)
+	if(active)
+		return
+	if(is_admin_level(bombturf.z) && !centcom_cancast)
+		to_chat(user, "<span class = 'warning'>The strike craft isn't authorized to fire near Central Command!</span>")
+		return
+	if(!authenticate(user, require_mindshield, authorized_special_roles, FALSE) && (ID_access_override && !authenticate(user, FALSE, FALSE, TRUE))) //if is not mindshielded, not an ERT/DS AND does not have high level CC access. naval officers have access_cent_commander, but not special role so we don't want to lock them out.
+		to_chat(user, "<span class = 'warning'>Unauthorized user detected.</span>")
+		return
+	active = TRUE
+	icon_state = initial(icon_state) + "_active"
+	item_state = initial(item_state) + "_active"
+	var/area/A = get_area(bombturf)
+	to_chat(user, "<span class = 'warning'>Calibrating GPS, remain still to continue with the strike.</span>")
+	//tell admins and ghosts we are initiating a strike
+	message_admins("[key_name_admin(user)] is initiating a Bluespace Artillery Strike at <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[bombturf.x];Y=[bombturf.y];Z=[bombturf.z]'>[A.name] (JMP)</a>") 
```

Comment:
Design:
The message_admins might be a bit much. Maybe only on it fully activating
Technically:
`ADMIN_VERBOSEJMP` or `ADMIN_COORDJMP` or such can be used for the jump.